src/libertem/udf/base.py

```diff
@@ -456,6 +471,36 @@ def set_meta(self, meta):
     def set_slice(self, slice_):
         self.meta.slice = slice_
 
+    def set_coordinates(self, slice_, ds_shape, roi):
+        o = slice_.origin
+        s = slice_.shape
+        sig_dims = s.sig.dims
+        start_idx = o[0]
+        end_idx = o[0] + s[0]
+        nav_shape = ds_shape[:-sig_dims]
+        if roi is None:
+            flat_nav_shape = tuple((int(np.prod(nav_shape)),))
+            self.meta.coordinates = [
+                np.unravel_index(
+                    np.ravel_multi_index(tuple((i,)), flat_nav_shape),
+                    nav_shape
+                ) for i in range(start_idx, end_idx)
+            ]
```

Comment:
Is it possible to maybe just call `unravel_index` / `ravel_multi_index` once, with `np.arange(start_idx, end_idx)` or similar? As this function is called for each tile it is a bit performance sensitive, so we have to be careful here to not introduce too much overhead. Maybe would be also a good benchmark to add, with large tiles as test data (for example whole partition)